GitUI/CommandsDialogs/FormFileHistory.cs

```diff
@@ -583,5 +604,74 @@ private void detectMoveAndCopyInThisFileToolStripMenuItem_Click(object sender, E
             detectMoveAndCopyInThisFileToolStripMenuItem.Checked = AppSettings.DetectCopyInAllOnBlame;
             UpdateSelectedFileViewers(true);
         }
+
+        private void CopyToClipboardToolStripMenuItem_DropDownOpened(object sender, EventArgs e)
+        {
+            var selectedRevisions = FileChanges.GetSelectedRevisions();
```

Comment:
Some redundency here... I would have introduced a method taking an `Action` like:

```csharp
        private void DoOnSelectedRevision(Action<GitRevision> ToDo)
        {
            var selectedRevisions = FileChanges.GetSelectedRevisions();
            if (selectedRevisions != null)
            {
                var r = selectedRevisions[0];
                if (r != null)
                {
                    ToDo(r);
                }
            }
        }
```

and used in all your methods

Reply:
Sure I’ll make that better. Thanks for your inputs.